generated/nidigital/nidigital/session.py

```diff
@@ -2871,14 +2871,31 @@ def write_source_waveform_site_unique(self, waveform_name, waveform_data):
         for site in waveform_data:
             if len(waveform_data[site]) != actual_samples_per_waveform:
                 raise ValueError('Mismatched length of waveforms. All must be the same length.')
-            if waveform_data[site].typecode != 'L':
-                raise ValueError('Wrong array element type. Must be unsigned 32 bit int ("L"), was {}'.format(waveform_data[site].typecode))
+            # Check the type by using string comparison so that we don't import numpy unecessarilly.
+            if str(type(waveform_data[site])).find("'numpy.ndarray'") != -1:
+                import numpy
+                if waveform_data[site].dtype == numpy.uint32:
+                    wfm = array.array('L', waveform_data[site])
+                else:
+                    raise TypeError("Unsupported dtype. Is {0}, expected {1}".format(waveform_data[site].dtype, numpy.int32))
+
+            elif isinstance(waveform_data[site], array.array):
+                if waveform_data[site].typecode == 'L':
+                    wfm = waveform_data[site]
+                else:
+                    raise ValueError('Wrong array element type. Must be unsigned 32 bit int ("L"), was {}'.format(waveform_data[site].typecode))
```

Comment:
Is it obvious what array we are referring to? Less ambiguous would be "waveform_data element type" or "waveform_data sample type"

Reply:
Updated description for `array.array` and `numpy.array` to mention "waveform_data array element type"

Change all raised exceptions to `TypeError` to 1) be consistent 2) seems more correct that `ValueError`